core/pkg/monitor/gpu_nvidia.go

```diff
@@ -3,220 +3,149 @@
 package monitor
 
 import (
+	"bufio"
+	"encoding/json"
 	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
 	"strings"
 	"sync"
-
-	"github.com/shirou/gopsutil/v4/process"
-
-	"github.com/NVIDIA/go-nvml/pkg/nvml"
+	"syscall"
+	"time"
 
 	"github.com/wandb/wandb/core/pkg/service"
 )
 
+// getCmdPath returns the path to the nvidia_gpu_stats program
+func getCmdPath() (string, error) {
+	ex, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+	exDirPath := filepath.Dir(ex)
+	exPath := filepath.Join(exDirPath, "nvidia_gpu_stats")
+
+	if _, err := os.Stat(exPath); os.IsNotExist(err) {
+		return "", err
+	}
+	return exPath, nil
+}
+
+// isRunning checks if the command is running by sending a signal to the process
+func isRunning(cmd *exec.Cmd) bool {
+	if cmd.Process == nil {
+		return false
+	}
+
+	process, err := os.FindProcess(cmd.Process.Pid)
+	if err != nil {
+		return false
+	}
+
+	err = process.Signal(syscall.Signal(0))
+	return err == nil
+}
+
 type GPUNvidia struct {
 	name     string
-	metrics  map[string][]float64
+	sample   map[string]any   // latest reading from nvidia_gpu_stats command
+	metrics  map[string][]any // all readings
```

Comment:
`metrics` grows indefinitely. If we need averages, we should track a total and a sum (using math/big). But also, do we really need an average? What about an exponential moving average---this would be numerically stable and only require tracking a single number per metric.

Reply:
the system monitor calls ClearMetrics after aggregation, but yea, agree - it's a previous design oversight that I just blindly copied over.
Let me get rid of averaging altogether, but in a next pr

Reply:
https://wandb.atlassian.net/browse/WB-20288